Source/WebKit/UIProcess/Downloads/DownloadProxyMap.cpp

```diff
@@ -69,10 +69,8 @@ void DownloadProxyMap::platformDestroy()
 }
 #endif
 
-DownloadProxy& DownloadProxyMap::createDownloadProxy(WebsiteDataStore& dataStore, WebProcessPool& processPool, const WebCore::ResourceRequest& resourceRequest, const FrameInfoData& frameInfo, WebPageProxy* originatingPage)
+DownloadProxy& DownloadProxyMap::createDownloadProxy(WebsiteDataStore& dataStore, Ref<API::DownloadClient>&& client, const WebCore::ResourceRequest& resourceRequest, const FrameInfoData& frameInfo, WebPageProxy* originatingPage)
 {
-    auto* legacyDownloadClient = processPool.legacyDownloadClient();
```

Comment:
Can we remove legacyDownloadClient() getter?

Reply:
Yes, it's removed in this patch.